File: 1455_check_if_a_word_occurs_as_a_prefix_of_any_word_in_a_sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string s, string p) {
        istringstream t(s);
        string u;
        for (int v = 1; t >> u; ++v) 
            if (u.find(p) == 0) return v;
        return -1;
    }
};

File: 1455_check_if_a_word_occurs_as_a_prefix_of_any_word_in_a_sentence.java
class Solution {
    public int isPrefixOfWord(String p, String q) {
        String[] arr = p.split(" ");
        int idx = 1;
        for (String w : arr) {
            if (w.startsWith(q)) return idx;
            idx++;
        }
        return -1;
    }
}
